### PR TITLE
[WIP] refactor(UT): addresses comments of PR #956

### PR DESCRIPTION
**Pull request by ashishranjan738**
Signed-off-by: Ashish Ranjan <ashishranjan738@gmail.com>

**What this PR does / why we need it**:
This PR addresses comments of PR https://github.com/openebs/maya/pull/956.

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/algorithm/cstorpoolselect/v1alpha1/select.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ type labelKey string
 const (
 	preferReplicaAntiAffinityLabel labelKey = "openebs.io/preferred-replica-anti-affinity"
 	replicaAntiAffinityLabel       labelKey = "openebs.io/replica-anti-affinity"
+	cstorPoolUIDLabelKey           labelKey = "cstorpool.openebs.io/uid"
```
Comment (AmitKumarDas) [inline pkg/algorithm/cstorpoolselect/v1alpha1/select.go]: Naming convention is changed. Why is `Key` introduced as a suffix?

File: pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go [hunk 2]
```diff
@@ -168,9 +164,9 @@ func TestNewSelection(t *testing.T) {
 
 func TestAntiAffinityFilter(t *testing.T) {
 	tests := map[string]struct {
-		CVRUid, poolUids, expectedUids []string
-		labelSelector                  string
-		expectError                    bool
+		CVRPoolUid, poolUids, expectedUids []string
```
Comment (vishnuitta) [inline pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go]: similar change in name might be required at other test functions (like `TestPreferredAntiAffinityFilter`) as well right?

Comment (AmitKumarDas) [reply pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go]:
```go
func fakeCVRListOk(uids ...string) cvrListFn {
  return func(namespace string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error) {
    l := &apis.CStorVolumeReplicaList{}
    for _, uid := range uids {
        l.Items = append(l.Items,
            apis.CStorVolumeReplica{
                ObjectMeta: metav1.ObjectMeta{
                    Labels: map[string]string{
                        string(cstorPoolUIDLabelKey): uid,
                    },
                },
            },
        )
    }
    return l, nil
  }
}

func fakeCVRListErr() cvrListFn {
  return func(namespace string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error) {
    return nil, errors.New("fake error")
  }
}

func TestAntiAffinityFilter(t *testing.T) {
    tests := map[string]struct {
        cvrList cvrListFn
        availablePools, expectedPools []string
    }{
        "Test 1": {
            cvrList:        fakeCVRListOk("uid 1"),
            availablePools: []string{"uid 1", "uid 2", "uid 3"},
            expectedPools: []string{"uid 1", "uid 2", "uid 3"},
        },
        "Test 2": {
            cvrList:        fakeCVRListOk("uid 4", "uid 2", "uid 7"),
            availablePools: []string{"uid 6"},
            expectedPools: []string{"uid 6"},
        },
        "Test 3": {
            cvrList:        fakeCVRListErr(),
            availablePools: []string{"uid 6"},
            expectedPools:  []string{},
        },
    }
    for name, test := range tests {
        t.Run(name, func(t *testing.T) {
            a := antiAffinityLabel{
                labelSelector: "should not be empty",
                cvrList: test.cvrList,
            }
            actual, _ := a.filter(test.providedPools)
            if len(test.expectedPools) != len(actual) {
                t.Fatalf("test %q failed: expected %v but got %v", name, test.expectedPools, actual)
            } 
            if len(output) != 0 && !reflect.DeepEqual(test.expectedPools, output) {
                t.Fatalf("test %q failed: expected %v but got %v", name, test.expectedPools, actual)
            }
        })
    }
}
```

File: pkg/algorithm/cstorpoolselect/v1alpha1/select.go (hunk 1 above)
Comment (ashishranjan738) [reply pkg/algorithm/cstorpoolselect/v1alpha1/select.go]: yeah updated it

File: pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go (hunk 2 above)
Comment (ashishranjan738) [reply pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go]: made the changes

**Comment by ashishranjan738** [comment]
we don't need it any more

**Pull request closed without merging**